### PR TITLE
Integrate vector into NamedVectors struct

### PR DESCRIPTION
**Pull request by IvanPleshkov**
Sparse vector index integration step. Propagate `Vector` type with sparse vector into `NamedVectors` struct.

### All Submissions:

* [ ] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [ ] Have you checked your code using `cargo clippy --all --all-features` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/segment/src/data_types/named_vectors.rs [hunk 1]
```diff
@@ -74,35 +165,62 @@ impl<'a> NamedVectors<'a> {
...
     }
 
-    pub fn into_default_vector(mut self) -> Option<Vec<VectorElementType>> {
-        self.map
-            .get_mut(DEFAULT_VECTOR_NAME)
-            .map(|src| std::mem::take(src).into_owned())
+    pub fn into_default_vector(mut self) -> Option<Vector> {
+        self.map.get_mut(DEFAULT_VECTOR_NAME).map(|src| match src {
+            CowValue::Dense(src) => Vector::Dense(std::mem::take(src).into_owned()),
+            CowValue::Sparse(src) => Vector::Sparse(std::mem::take(src).into_owned()),
```
Comment (agourlay) [inline lib/segment/src/data_types/named_vectors.rs:171]: In practice we do not want to support anonymous sparse vector.
Is this conversion mandatory?

**Review by agourlay — COMMENTED** [review]
There are really a lot of new conversions. it is a bit difficult to keep track of it TBH.
Maybe we can review together.

File: lib/segment/src/data_types/named_vectors.rs (hunk 1 above)
Comment (IvanPleshkov) [reply lib/segment/src/data_types/named_vectors.rs]: marked as todo to remove this functions

File: lib/segment/src/data_types/named_vectors.rs [hunk 2]
```diff
@@ -74,35 +113,63 @@ impl<'a> NamedVectors<'a> {
...
-    pub fn into_owned_map(self) -> HashMap<String, Vec<VectorElementType>> {
+    pub fn into_owned_map(self) -> HashMap<String, Vector> {
         self.map
             .into_iter()
-            .map(|(k, v)| (k.into_owned(), v.into_owned()))
+            .map(|(k, v)| {
+                (
+                    k.into_owned(),
+                    match v {
+                        CowValue::Dense(src) => Vector::Dense(src.into_owned()),
```
Comment (agourlay) [inline lib/segment/src/data_types/named_vectors.rs:131]: can we use the new `to_owned` implementation for `CowValue` there?

Comment (IvanPleshkov) [reply lib/segment/src/data_types/named_vectors.rs]: Fixed!

File: lib/segment/src/data_types/vectors.rs [hunk 3]
```diff
@@ -143,8 +143,22 @@ pub fn default_vector(vec: Vec<VectorElementType>) -> NamedVectors<'static> {
     NamedVectors::from([(DEFAULT_VECTOR_NAME.to_owned(), vec)])
 }
 
+pub fn default_sparse_vector(vec: SparseVector) -> NamedVectors<'static> {
```
Comment (agourlay) [inline lib/segment/src/data_types/vectors.rs:146]: we don't default for sparse

**Pull request merged**